crates/catalog/rest/src/catalog.rs

```diff
@@ -956,6 +964,39 @@ mod tests {
         );
     }
 
+    #[tokio::test]
+    async fn test_oauth_with_auth_url() {
+        let mut server = Server::new_async().await;
+        let config_mock = create_config_mock(&mut server).await;
+
+        let mut auth_server = Server::new_async().await;
+        let auth_server_path = "/some/path";
+        let oauth_mock = create_oauth_mock_with_path(&mut auth_server, auth_server_path).await;
```

Comment:
To explain the unit test, we first create a mock response for our OAuth Server. Then in line 978 - 981, set `rest.authorization-url` to use the OAuth Server (instead of default server). Finally, we verify the token is correct returned from our OAuth Server.

Reply:
Do we need to assert the `rest.authorization-url` using .match?

Reply:
No, because we already assert `rest.authorization-url` implicitly. If `rest.authorization-url` is not honored, we won't be able to get `token` from the mocked OAuth server (using same url) 🙂.